### ISSUE TITLE
'Powered by Esri' missing in map attribution

### ISSUE THREAD

**Issue by ItaiEinhorn** [Esri/esri-leaflet-vector#133]
### Describe the bug

Other Esri leaflet map layers support the 'Powered by Esri' attribution
`onAdd: function (map) {
    // include 'Powered by Esri' in map attribution
    setEsriAttribution(map);
...`
Vector layers does not support this behavior

Requesting to add the 'setEsriAttribution' call to the 'onAdd' function in the vector layers

### Reproduction

Tiled map layer:
![image](https://user-images.githubusercontent.com/66256027/177318530-84f31c35-19d2-4213-b94b-d13ab0154d4d.png).
Vector Layer:
![image](https://user-images.githubusercontent.com/66256027/177319723-0d9f68ba-2917-40b1-add6-f7051337aa7f.png)

### System Info

```shell
Leaflet version: `v1.7.1`. 
Esri Leaflet version: `v3.0.7`.
Esri Leaflet Vector version: `v3.1.2`.
```

**Comment by ItaiEinhorn** [comment]
@gavinr Thanks for the response
I took the source code from the same page in your screenshot, the only change:
```
L.esri.Vector.vectorBasemapLayer("3e1a00aeae81496587988075fe529f71", {
        url: 'https://basemaps.arcgis.com/arcgis/rest/services/OpenStreetMap_v2/VectorTileServer',
        apikey: apiKey
      }).addTo(map);
```

This map doesn't show the 'Powered by Esri'

![image](https://user-images.githubusercontent.com/66256027/177569197-c8309929-1ae1-43c7-89a6-af19052801c4.png)

**Comment by gavinr** [comment]
Are you able to use this?

```js
L.esri.Vector.vectorBasemapLayer("OSM:Standard", {
  apikey: apiKey
}).addTo(map);
```

That way includes the "Powered by Esri".

**Comment by ItaiEinhorn** [comment]
@gavinr Unfortunately i can't, In my use case the basemaps are fetched dynamically from the user's gallery

**Comment by gavinr** [comment]
In that case, you can pass the `attribution` parameter, like this:

```js
L.esri.Vector.vectorBasemapLayer("3e1a00aeae81496587988075fe529f71", {
      apiKey: apiKey,
      attribution: 'Powered by <a href="https://esri.com">Esri</a>'
}).addTo(map);
```

**Comment by ItaiEinhorn** [comment]
@gavinr That's a nice workaround,
However, this way we'll lose the current attribution. I was expecting from esri-leaflet-vector library to add the 'Powered by Esri' the same way it does with named basemaps, am i missing something?

**Comment by gavinr** [comment]
>  I was expecting from esri-leaflet-vector library to add the 'Powered by Esri' the same way it does with named basemaps

If you add a vector tile layer via an item ID generally, it should not necessarily add 'Powered by Esri'. In your case you are adding a vector tile layer via an item ID that happens to be one of the Esri basemap layers.  I'm not quite sure if we should or can handle that situation.

Right now it does correctly add the "accessInformation" from the item info:
https://prof-services.maps.arcgis.com/sharing/rest/content/items/3e1a00aeae81496587988075fe529f71?f=json

![image](https://user-images.githubusercontent.com/209355/178356355-390517e9-3941-4d01-a8ec-b78549cbad05.png)

... which I think is correct.

**Comment by ItaiEinhorn** [comment]
@gavinr Why is this different from other layers? for example: RasterLayer has the 'setEsriAttribution' call in the 'onAdd' event (same with other layers in that library) but for VectorBasemapLayer it calls 'setEsriAttribution' only on '_setupAttribution' callback only if id isn't 32 characters.
This is inconsistent and confusing behavior.
If you don't necessarily add 'Powered by Esri' in esri-leaflet-vector then we should have the same behavior in other types of base map layers

**Comment by gavinr** [comment]
Thanks for the input @ItaiEinhorn. I think you're right - every standard Esri Leaflet layer type adds "Powered by Esri" except for this layer in this very specific case ([demo](https://codepen.io/gavinr/full/ZExpKjx)). Given that, I agree with you that this layer should be consistent and add "Powered by Esri" in all use cases.

PRs to update this are welcome!

**Comment by shawnmgoulet** [comment]
I've submitted a PR referenced above (#135) that seeks to resolve this issue. @ItaiEinhorn @gavinr

**Comment by ItaiEinhorn** [comment]
@gavinr @shawnmgoulet Thank you guys,

Any idea when a new version will be published?

**Comment by gavinr** [comment]
The fix for this (#135) was released in [v3.1.4](https://github.com/Esri/esri-leaflet-vector/releases/tag/v3.1.4).
